tests/unit/decorators-test.ts

```diff
@@ -76,6 +77,72 @@ module('Unit | decorators', function() {
     assert.equal(subject.get('d.last.value'), 34);
   });
 
+  test('Basic decorators functionality (using wrapper for TS support)', function(assert) {
+    assert.expect(5);
+
+    class TestSubject extends EmberObject {
+      @task
+      doStuff = task(function*() {
+        yield;
+        return 123;
+      });
+
+      @restartableTask
+      a = task(function*() {
+        yield;
+        return 456;
+      });
+
+      @keepLatestTask
+      b = task(function*() {
+        yield;
+        return 789;
+      });
+
+      @dropTask
+      c = task(function*() {
+        yield;
+        return 12;
+      });
+
+      @enqueueTask
+      d = task(function*() {
+        yield;
+        return 34;
+      });
+    }
+
+    let subject!: TestSubject;
+    run(() => {
+      subject = TestSubject.create();
+      subject.get('doStuff').perform();
+      subject.get('a').perform();
+      subject.get('b').perform();
+      subject.get('c').perform();
+      subject.get('d').perform();
+    });
+    // eslint-disable-next-line ember/new-module-imports
+    if (/^2\./.test(Ember.VERSION)) {
+      // Have access computed properties with .get w/ Ember 2.x
+      // @ts-ignore
+      assert.equal(subject.get('doStuff.last.value'), 123);
+      // @ts-ignore
+      assert.equal(subject.get('a.last.value'), 456);
+      // @ts-ignore
+      assert.equal(subject.get('b.last.value'), 789);
+      // @ts-ignore
+      assert.equal(subject.get('c.last.value'), 12);
+      // @ts-ignore
+      assert.equal(subject.get('d.last.value'), 34);
```

Comment:
When https://github.com/machty/ember-concurrency/pull/363 lands, we can remove this ugly special-casing for the 2.18 try case and just use, e.g.
```ts
assert.equal(subject.get('doStuff').get('last')?.value, 123);
```

Reply:
Couldn't resist: https://github.com/machty/ember-concurrency-decorators/pull/76/commits/b5dbcb40a0b86dfbf76bf96a10240e5befda1329